nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/spi/PayaraConfig.java

```diff
@@ -225,11 +224,13 @@ protected ConfigValueImpl getConfigValue(String propertyName, String cacheKey, L
         final String entryKey = cacheKey + (defaultValue != null ? ":" + defaultValue : "")  + ":" + (entryTTL / 1000) + "s";
         final long now = currentTimeMillis();
 
+        boolean isExpansionEnabled = isExpansionEnabled(propertyName);
+
         return cachedValuesByProperty.compute(entryKey, (key, entry) -> {
```

Comment:
I don't think we can keep this call to compute. Because when expansion *is* enabled, other entries still need to get resolved during computation and break the contract of `compute` method.

Reply:
You are right, searchConfigSources() could cause another recursion, e.g. breaking compute contract. Fixed in new version.